### ISSUE TITLE
Setup Docker building and publishing for existing source trees on merge into main

### ISSUE THREAD

**Issue by ashketchupppp** [Jemila-Abdulai/software-engineering-2#4]
Blocked by #2 

We should create new GitHub actions to trigger a Docker build and publish to Docker hub for each source tree, triggered for every commit on the main branch (i.e. when a PR is merged)
